pennylane/devices/tests/test_compare_default_qubit.py

```diff
@@ -145,57 +145,70 @@ def circuit(weights):
-        dev_def = qml.device("default.qubit", wires=n_wires)
-
-        if dev.name == dev_def.name:
-            pytest.skip("Device is default.qubit.")
 
-        if not dev.analytic:
-            pytest.skip("Device is in non-analytical mode.")
+        tape_mode_active = qml.tape_mode_active()
+
+        if not tape_mode_active:
+            qml.enable_tape()
+
+        try:
+            n_wires = 4
+            dev = device(n_wires)
+            dev_def = qml.device("default.qubit", wires=n_wires)
+
+            if dev.name == dev_def.name:
+                pytest.skip("Device is default.qubit.")
+
+            if not dev.analytic:
+                pytest.skip("Device is in non-analytical mode.")
+
+            gates = [
+                qml.PauliX,
+                qml.PauliY,
+                qml.PauliZ,
+                qml.S,
+                qml.T,
+                qml.RX,
+                qml.RY,
+                qml.RZ,
+                qml.Hadamard,
+                qml.Rot,
+                qml.CRot,
+                qml.Toffoli,
+                qml.SWAP,
+                qml.CSWAP,
+                qml.U1,
+                qml.U2,
+                qml.U3,
+                qml.CRX,
+                qml.CRY,
+                qml.CRZ,
+            ]
 
-        gates = [
-            qml.PauliX,
-            qml.PauliY,
-            qml.PauliZ,
-            qml.S,
-            qml.T,
-            qml.RX,
-            qml.RY,
-            qml.RZ,
-            qml.Hadamard,
-            qml.Rot,
-            qml.CRot,
-            qml.Toffoli,
-            qml.SWAP,
-            qml.CSWAP,
-            qml.U1,
-            qml.U2,
-            qml.U3,
-            qml.CRX,
-            qml.CRY,
-            qml.CRZ,
-        ]
-
-        layers = 3
-        np.random.seed(1967)
-        gates_per_layers = [np.random.permutation(gates).numpy() for _ in range(layers)]
-
-        def circuit():
-            """4-qubit circuit with layers of randomly selected gates and random connections for
-            multi-qubit gates."""
+            layers = 3
             np.random.seed(1967)
-            for gates in gates_per_layers:
-                for gate in gates:
-                    params = list(np.pi * np.random.rand(gate.num_params))
-                    gate(
-                        *params,
-                        wires=np.random.choice(range(n_wires), size=gate.num_wires, replace=False)
-                    )
-            return qml.expval(qml.PauliZ(0))
-
-        qnode_def = qml.QNode(circuit, dev_def)
-        qnode = qml.QNode(circuit, dev)
-
-        assert np.allclose(qnode(), qnode_def(), atol=tol(dev.analytic))
+            gates_per_layers = [np.random.permutation(gates).numpy() for _ in range(layers)]
+
+            def circuit():
+                """4-qubit circuit with layers of randomly selected gates and random connections for
+                multi-qubit gates."""
+                np.random.seed(1967)
```

Comment:
(Minor:) Not entirely in scope (as this was not changed), but do we need this seeding and the one on line 189 too? 🤔 
```suggestion
```